src/main/java/org/mamute/dao/QuestionDAO.java

```diff
@@ -272,5 +273,10 @@ public Question fromCommentId (Long id) {
 	}
 
 
+	public Question questionWith(Attachment attachment) {
+		return (Question) session.createQuery("from Question where :a in elements(attachments)")
+				.setParameter("a", attachment)
+				.uniqueResult();
+	}
```

Comment:
We can make this generic if we use an interface like I suggested above. (something like FlaggableDAO)